lib/elixir/lib/task.ex

```diff
@@ -338,13 +341,17 @@ defmodule Task do
   end
 
   @doc """
-  Receives a group of tasks and a message and finds
+  Takes a list of tasks and a message and finds
   a task that matches the given message.
 
-  This function returns a tuple with the returned value
-  in case the message matches a task that exited with
-  success alongside the matching task. It returns `nil`
-  if no task was found. It exits if the task has failed.
+  If the message matches a task that finished normally,
+  this function returns a tuple with the task's
+  result and `Task` struct.
+
+  If the message matches a task that terminated before
+  sending its result back, this function will exit.
```

Comment:
When `Task.find` was introduced, we didn't have `async_nolink` and changing it would be backwards incompatible. We were also not sure if the `Task.find` would stick. Today, there are two ways of using tasks:

1. You want it to compute something in asynchronously and you want it to go until the end (async + await)

2. You want to start some computation that you may not necessarily want to wait until it finishes (async + yield + shutdown)

At this point, I am actually wondering if we shouldn't just kill `Task.find`. If you have a server, you can easily use `Task.start_link` and send a message back. Today, the task message format is documented and unlikely to change, so you can also check for it explicitly.

Thoughts? Of course, it would have to be on 1.3.